src/Inputs/Slider.tsx

```diff
@@ -1,79 +1,384 @@
-    }
-    &::-webkit-slider-thumb {
-        box-shadow: 1px 1px 1px #000000, 0px 0px 1px #0d0d0d;
-        border: 1px solid #000000;
-        height: 16px;
-        width: 16px;
-        border-radius: 50%;
-        background: ${(props): string => props.buttonColor || '#ffffff'};
-        cursor: pointer;
-        -webkit-appearance: none;
-        margin-top: -5px;
-    }
-    &:focus::-webkit-slider-runnable-track {
-        background: #367ebd;
-    }
-    &::-moz-range-track {
+}): React.ReactElement => {
+    // DOM Elements
+    const barRef = useRef<HTMLDivElement>(null);
+    const selectedBarRef = useRef<HTMLDivElement>(null);
+    const marksBarRef = useRef<HTMLDivElement>(null);
+
+    // Knob Positions
+    const [rightKnobPosition, setRightKnobPosition] = useState(
+        values.rightValue,
+    );
+
+    const [leftKnobPosition, setLeftKnobPosition] = useState(values.leftValue);
+
+    const [isRightKnobDragging, setIsRightKnobDragging] = useState(false);
+    const [isLeftKnobDragging, setIsLeftKnobDragging] = useState(false);
+
+    const maxAndMinDifference = useMemo((): number => {
+        return max - min;
+    }, [max, min]);
+
+    // Translate a value to Pixel
+    const translateToPixels = (theValue: number): number => {
+        const pixelTranslator =
+            (barRef.current?.clientWidth as number) / maxAndMinDifference;
+        return (theValue - min) * pixelTranslator;
+    };
+
+    // Rounding up/down to steps
+    const roundToSteps = (theValue: number): number => {
+        return (
+            Math.round(
+                ((theValue * maxAndMinDifference) /
+                    (barRef.current?.clientWidth as number) +
+                    min) /
+                    step,
+            ) * step
+        );
+    };
+
+    useLayoutEffect((): void => {
+        onChange({
+            values: {
+                leftValue: leftKnobPosition,
+                rightValue: rightKnobPosition,
+            },
+        });
+    }, [rightKnobPosition, leftKnobPosition]);
+
+    const onMouseMove = (theevent: MouseEvent): void => {
+        // clicked on pure position
+        const newLeft =
+            theevent.clientX -
+            (barRef.current as HTMLElement).getBoundingClientRect().left;
+
+        // new position based on steps
+        const newPosition = roundToSteps(newLeft);
+
+        // setting the positions
+        if (newLeft < (barRef.current?.clientWidth as number) && newLeft >= 0) {
+            if (
+                (isRightKnobDragging && newPosition > leftKnobPosition) ||
+                !hasTwoKnobs
+            ) {
+                setRightKnobPosition(newPosition);
+            } else if (isLeftKnobDragging && newPosition < rightKnobPosition) {
+                setLeftKnobPosition(newPosition);
+            }
+        }
+    };
+
+    const onMouseUp = (): void => {
+        document.removeEventListener('mouseup', onMouseUp);
```

Comment:
should probably add setIsRightKnobDragging(false) and setIsLeftKnobDragging(false) here